Microsoft.Toolkit.Uwp/Helpers/BackgroundTaskHelper.cs

```diff
@@ -0,0 +1,107 @@
+        /// <param name="conditions">Optional conditions for the background task to run with</param>
+        /// <returns>Background Task that was registered with the system</returns>
+        public static async Task<BackgroundTaskRegistration> RegisterAsync(Type backgroundTaskType, IBackgroundTrigger trigger, bool enforceConditions = true, params IBackgroundCondition[] conditions)
+        {
+            // Check if the task is already registered.
+            if (IsBackgroundTaskRegistered(backgroundTaskType) == true)
+            {
+                return GetBackgroundTask(backgroundTaskType) as BackgroundTaskRegistration;
+            }
+
+            // Verify access
+            BackgroundAccessStatus taskRequired = await BackgroundExecutionManager.RequestAccessAsync();
+
+            if (taskRequired == BackgroundAccessStatus.Denied)
+            {
+                throw new InvalidOperationException("Background access is denied!");
+            }
+
+            // Get details about the background task
+            string backgroundTaskName = backgroundTaskType.Name;
+            string backgroundTaskEntryPoint = backgroundTaskType.FullName;
+
+            // build the background task
+            BackgroundTaskBuilder builder = new BackgroundTaskBuilder();
+            builder.Name = backgroundTaskName;
+            builder.TaskEntryPoint = backgroundTaskEntryPoint;
+            builder.CancelOnConditionLoss = enforceConditions;
+
+            // add the conditions if we have them.
+            conditions?.ToList().ForEach(builder.AddCondition);
+
+            builder.SetTrigger(trigger);
+
+            // Register it
+            BackgroundTaskRegistration registered = builder.Register();
+
+            return registered;
+        }
+
+        /// <summary>
+        /// Unregister a background task
+        /// </summary>
+        /// <param name="backgroundTaskType">The type of the background task</param>
+        public static void Unregister(Type backgroundTaskType)
+        {
+            if (IsBackgroundTaskRegistered(backgroundTaskType) == false)
+            {
+                return;
+            }
+
+            IBackgroundTaskRegistration toBeUnregister = BackgroundTaskRegistration.AllTasks.Where(t => t.Value.Name == backgroundTaskType.Name).Select(t => t).FirstOrDefault().Value;
+
+            toBeUnregister?.Unregister(true);
+        }
+
+        /// <summary>
+        /// Get the registered background task of the given type.
+        /// </summary>
+        /// <param name="backgroundTaskType">Type of the background task class</param>
+        /// <returns>Background task if there is such background task registered. Otherwise, null</returns>
+        public static IBackgroundTaskRegistration GetBackgroundTask(Type backgroundTaskType)
+        {
+            return BackgroundTaskRegistration.AllTasks.FirstOrDefault(t => t.Value.Name == backgroundTaskType.Name).Value;
```

Comment:
This will throw a NullReferenceException, since you're doing FirstOrDefault, and then accessing a property on the returned result.

Throw in the "?" symbol so that it'll automatically return null :) so with a one-character change, this is fixed:

    return BackgroundTaskRegistration.AllTasks.FirstOrDefault(t => t.Value.Name == backgroundTaskType.Name)?.Value;